zipline/testing/core.py

```diff
@@ -1451,7 +1451,7 @@ def __enter__(self):
                 self._data,
             )
             return self._reader_cls(tmpdir.path)
-        except:
+        except BaseException:  # Gotta catch 'em all
```

Comment:
I almost don't even think this is a reasonable change. If you actually want to catch all exceptions `except:` does that and is pretty clear about it.

Reply:
Could we move to `except Exception:` and assume `MemoryError` and the like are unrecoverable?

Reply:
This code doesn't recover, it calls calls an `__exit__` and then re-raises. If you are re-raising I think it is fine to catch everything.

Reply:
You can get a memory error for a lot of reasons, `np.arange(2 ** 40)` will memory error but you still have plenty of memory when you need to clean up. Even recursion error is likely fine because once you make it back to the except handler you have unwound all the frames created in the body of the try so you have spare frames to cleanup.

Reply:
It looks like catching only `Exception` would result in lingering temp files if sigint or sigterm is received here, so I think `BaseException` is the right choice. (`MemoryError` is actually a subclass of `Exception`.)

I agree that it is abundantly clear to you and me what `except` does, but whenever I encounter it in someone else's code, I wonder whether they really meant it (as we seem to not have at two other locations in this PR).

I personally think `# noqa` looks/feels a little grosser than `BaseException`, but they're basically equivalent to me.